### PR TITLE
refactor(gsdk): use `Into<Option<T>>` instead of `AsOption`

### PR DESCRIPTION
**Pull request by clearloop**
Refactor of  #4022.
Ref  #4009

@gear-tech/dev

### CODE REVIEW

File: gsdk/src/api.rs [hunk 1]
```diff
@@ -37,8 +37,8 @@ pub struct Api {
 
 impl Api {
     /// Create new API client.
-    pub async fn new(url: impl AsOption<str>) -> Result<Self> {
-        Self::new_with_timeout(url, None).await
+    pub async fn new(url: impl Into<Option<String>>) -> Result<Self> {
```
Comment (breathx) [inline gsdk/src/api.rs:40]: &str as well

**Review by breathx — APPROVED** [review]

File: gsdk/src/api.rs (hunk 1 above)
Comment (StackOverflowExcept1on) [reply gsdk/src/api.rs]: mb `AsRef<str>`?

Comment (clearloop) [reply gsdk/src/api.rs]: the problem here is `String` is most frequently used for this interface ...

```
// impl Into<Option<String>>
let api = Api::new(node.ws());

// impl Into<Option<&str>>
let api = Api::new(node.ws().as_str())

// impl Into<Option<T>> where T: AsRef<str>
let api = Api::new::<String>(node.ws())
```

Comment (clearloop) [reply gsdk/src/api.rs]: there problem here is, `String` is the most frequently used case

```rust
// impl Into<Option<String>>
let api = Api::new(node.ws());
let api = Api::new(Some("".to_string()))
let api = Api::new(None)

// impl Into<Option<&str>>
let api = Api::new(node.ws().as_str())
let api = Api::new(Some(""))
let api = Api::new(None)

// impl<T: AsRef<str>> Into<Option<T>>
let api = Api::new(node.ws())
let api = Api::new::<String>(Some("".to_string()))
let api = Api::new::<String>(None)
```

Comment (StackOverflowExcept1on) [reply gsdk/src/api.rs]: makes sense if rust cannot infer type

**Review by StackOverflowExcept1on — APPROVED** [review]

**Pull request merged**